main/pipeline/pipeline.cc

```diff
@@ -78,25 +78,66 @@ class CFGCollectorAndTyper {
     }
 };
 
-string fileKey(core::GlobalState &gs, core::FileRef file) {
-    auto path = file.data(gs).path();
+string fileKey(const core::File &file, string_view subtype = ""sv) {
```

Comment:
There's a lot of additional complexity here, which, if we only store FileHashes in global state won't be necessary